contrib/exporters/secadvisor/mod/acc_rej_action.go

```diff
@@ -0,0 +1,218 @@
+ * You may obtain a copy ofthe License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specificlanguage governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package mod
+
+import (
+	"time"
+
+	"github.com/skydive-project/skydive/common"
+	"github.com/skydive-project/skydive/flow"
+	"github.com/spf13/viper"
+)
+
+type FlowInfoStatus int32
+
+const (
+	FlowInfoOneEntry   FlowInfoStatus = 1
+	FlowInfoTwoEntries FlowInfoStatus = 2
+	FlowInfoOldFlow    FlowInfoStatus = 3
+)
+
+// time for cleanup of old connections
+const timeoutPeriod = 20
+
+type accRejFlowInfo struct {
+	Status                   FlowInfoStatus
+	UUID1                    string
+	UUID2                    string
+	MetricReported           flow.FlowMetric
+	LastUpdateMetricReported flow.FlowMetric // FIXME - we might not need this
+	Saw1                     bool
+	Saw2                     bool
+	Flow1                    *flow.Flow
+	Flow2                    *flow.Flow
+}
+
+type accRejAction struct {
+	// table of saved flows
+	accRejState map[string]*accRejFlowInfo
+	timeOfLastCleanup	int64
+}
+
+// NewAction - creates data structures needed for vpc Accept/Reject based on measurements flows at 2 points
+func NewAction(cfg *viper.Viper) (interface{}, error) {
+	accRejState := make(map[string]*accRejFlowInfo)
+	now := time.Now()
+	secs := now.Unix()
+	accRej := &accRejAction{
+		accRejState:  accRejState,
+		timeOfLastCleanup: secs,
+	}
+	return accRej, nil
+}
+
+// initAction - initialize internal data structure for this round
+func (t *accRejAction) initAction() {
+	// reset the entries in the table for this round
+	for _, ars := range t.accRejState {
+		ars.Saw1 = false
+		ars.Saw2 = false
+		ars.Status = FlowInfoOldFlow
+	}
+}
+
+// updateInfo - go over all the flows and fill in our internal data structure with relevant info
+func (t *accRejAction) updateInfo(fl []*flow.Flow) {
+	for _, f := range fl {
+		L3TrackingID := f.GetL3TrackingID()
+		flowInf, ok := t.accRejState[L3TrackingID]
+		if !ok {
+			// this flow is not yet in our table
+			t.accRejState[L3TrackingID] = &accRejFlowInfo{
+				UUID1:   f.UUID,
+				Status:  FlowInfoOneEntry,
+				Saw1:    true,
+				Flow1:   f,
+			}
+			continue
+		} else if flowInf.Status == FlowInfoOneEntry {
+			if flowInf.UUID1 != f.UUID {
+				flowInf.UUID2 = f.UUID
+				flowInf.Saw2 = true
+				flowInf.Flow2 = f
+				flowInf.Status = FlowInfoTwoEntries
+			} else {
+				// got duplicate from same flow; need to indicate reject
+				// or accumulate the stats until next time
+				flowInf.Saw1 = true
+				flowInf.Flow1 = f
+				continue
+			}
+		} else { // normal case
+			if flowInf.UUID1 == f.UUID {
+				flowInf.Flow1 = f
+				flowInf.Saw1 = true
+			} else {
```

Comment:
You must ensure that `if flowInt.UUID2 == f.UUID`

Reply:
OK. If UUID2 is not equal to UUID, then we have at least three reports on the same flow. This breaks the assumption of the setup. In this case we'll have to simply discard the entire entry for that flow.